src/App.test.tsx

```diff
@@ -60,8 +60,35 @@ describe("<App />", () => {
     expect(firstSuggestion).toHaveTextContent("Maczugi keczupowe");
     userEvent.click(firstSuggestion);
 
+    expect(await screen.findByText(/Treść problemu/i)).toBeInTheDocument();
+  });
+
+  it("navigates from detailed page to problems page after clicking on 'return to problems list' button", async () => {
+    testRenderer(<App />);
+    const problem = (await screen.findByText(
+      "Ciągi matematyczne"
+    )) as HTMLInputElement;
+    userEvent.click(problem);
+
+    const returnButton = (await screen.findByText(
+      "Powrót do listy problemów"
+    )) as HTMLInputElement;
+    userEvent.click(returnButton);
+
+    expect(
+      await screen.findByText(/Szukaj problemów matematycznych/i)
+    ).toBeInTheDocument();
+  });
+
+  it("shows problem extension after clicking on suggestion ", async () => {
```

Comment:
Actually it shows problem detailed page, where problem extension is present along with other elements ;)